src/neo/Trie/MPT/MPTTrie.cs

```diff
@@ -0,0 +1,374 @@
+                            Next = child,
+                        };
+                        node = newNode;
+                        db.Put(node);
+                        return true;
+                    }
+                case HashNode hashNode:
+                    {
+                        if (hashNode.IsEmptyNode)
+                        {
+                            return true;
+                        }
+                        node = Resolve(hashNode.Hash);
+                        return tryDelete(ref node, path);
+                    }
+                default:
+                    return false;
+            }
+        }
+
+        public byte[] GetRoot()
+        {
+            return this.root.GetHash();
+        }
+
+        public Dictionary<byte[], byte[]> GetProof(byte[] path)
+        {
+            var dict = new Dictionary<byte[], byte[]> { };
+            path = path.ToNibbles();
+            getProof(ref root, path, dict);
+            return dict;
+        }
+
+        private void getProof(ref MPTNode node, byte[] path, Dictionary<byte[], byte[]> dict)
+        {
+            switch (node)
+            {
+                case ValueNode valueNode:
+                    {
+                        if (path.Length == 0)
+                        {
+                            dict.Add(valueNode.GetHash(), valueNode.Encode());
+                        }
+                        break;
+                    }
+                case HashNode hashNode:
+                    {
+                        if (hashNode.IsEmptyNode) break;
+                        node = Resolve(hashNode.Hash);
+                        dict.Add(node.GetHash(), node.Encode());
+                        getProof(ref node, path, dict);
+                        break;
+                    }
+                case FullNode fullNode:
+                    {
+                        dict.Add(fullNode.GetHash(), fullNode.Encode());
+                        if (path.Length == 0)
+                        {
+                            getProof(ref fullNode.Children[16], path, dict);
+                        }
+                        else
+                        {
+                            getProof(ref fullNode.Children[path[0]], path.Skip(1), dict);
+                        }
+                        break;
+                    }
+                case ShortNode shortNode:
+                    {
+                        var prefix = shortNode.Key.CommonPrefix(path);
+                        if (prefix.Length == shortNode.Key.Length)
+                        {
+                            dict.Add(shortNode.GetHash(), shortNode.Encode());
+                            getProof(ref shortNode.Next, path.Skip(prefix.Length), dict);
+                        }
+                        break;
+                    }
+            }
+        }
+
+        public void Commit()
+        {
+            db.PutRoot(GetRoot());
+        }
+    }
+}
```

Comment:
A blank line should be added in the end of the file, otherwise the dotnet-format won't pass.